File: career/src/main/java/com/infosis/nexus/career/service/JobOpeningService.java
package com.infosis.nexus.career.service;

import com.infosis.nexus.career.dto.*;
import java.util.List;

public interface JobOpeningService {
    JobOpeningResponse create(JobOpeningRequest req);
    JobOpeningResponse getById(Long id);
    List<JobOpeningResponse> getAll();
    JobOpeningResponse update(Long id, JobOpeningRequest req);
    void delete(Long id);
    List<JobMatchResponse> getMatchesForEmployee(Long employeeId);
}

File: frontend/src/services/jobOpeningService.js
import api from './api';

const BASE = '/api/career/job-openings';

const jobOpeningService = {
    getAll: () => api.get(BASE),
    getById: (id) => api.get(`${BASE}/${id}`),
    create: (data) => api.post(BASE, data),
    update: (id, data) => api.put(`${BASE}/${id}`, data),
    delete: (id) => api.delete(`${BASE}/${id}`),
    getMatchesForEmployee: (employeeId) => api.get(`${BASE}/matches/employee/${employeeId}`),
};

export default jobOpeningService;